### PR TITLE
Implement function names with dots in signatures ##anal

### PR DESCRIPTION
**Pull request by oxcabe**
**Your checklist for this pull request**

- [x] I've read the guidelines for contributing to this repository
- [x] I made sure to follow the project's coding style
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the radare2 book with the relevant information (if needed)

**Detailed description**

Merging this Pull Request would allow setting function names containing dots in function signatures.

Current state of this PR is:
- [x] User can set function signatures with dots in the function name using `afs`.
- [ ] Signatures with dots in function name are stored properly in the anal/types SDB.
The current issue (addressed by the second milestone) now is that functions with dots don't get the function type and arguments correctly.

~~In order to fix this, we might need some kind of structure that annotates where the dots were replaced by underscores in the function name, so it can find the proper keys in SDB when querying it.~~. 

Fix functions not getting right arguments in a **new** issue and PR.

**Test plan**
Include a test that asserts if adding dots in the function name of a signature works as intended.

**Closing issues**
Closes https://github.com/radareorg/radare2/issues/15954

### CODE REVIEW

**Comment by radare** [comment]
maybe we should fix this in RAnalFunction.setName() instead.. but this wont let users to use dots in the function names in r2.. which imho it should be valid char, but it is not valid for tcc. so maybe better to fix tcc instead

**Comment by oxcabe** [comment]
So, allow dots in function names in TCC?

**Comment by radare** [comment]
yep. hopefully it wont break too many things :D what do you think xvilka? ret2libc?

**Comment by ret2libc** [comment]
Please do use the PR template next time.

**Comment by oxcabe** [comment]
Description is updated now, using the PR template

**Comment by thestr4ng3r** [comment]
> maybe we should fix this in RAnalFunction.setName() instead.. but this wont let users to use dots in the function names in r2.. which imho it should be valid char, but it is not valid for tcc. so maybe better to fix tcc instead

I agree with this, we shouldn't limit the zignatures functionality by only the limitations of the parser. We also considered switching from tcc to tree sitter before, which would let us change the parser more easily.

**Comment by radare** [comment]
better approach but still red

File: shlr/tcc/tccpp.c [hunk 1]
```diff
@@ -2353,18 +2353,29 @@ static inline void next_nomacro1(void)
...
+	case '_': case '.':
 parse_ident_fast:
 		p1 = p;
 		h = TOK_HASH_INIT;
 		h = TOK_HASH_FUNC (h, c);
 		p++;
 		for (;;) {
 			c = *p;
-			if (!isidnum_table[c - CH_EOF]) {
+			if (!isidnum_table[*p - CH_EOF]) {
```
Comment (radare) [inline shlr/tcc/tccpp.c]: this is the same, isnt?

**Pull request merged**